### ISSUE TITLE
Add more logging of exceptions in backends

### ISSUE THREAD

**Issue by Skylion007** [fsspec/filesystem_spec#1642]
A lot of backends can throw exceptions that are swallowed by fsspec, we should at least log these errors in debug or info level using the builtin loggers with exc_info=True to improve the debugging experience of the library.

**Comment by martindurant** [comment]
Agree, we could do with more logging throughout the library, and including exception info where events are due to errors sounds very reasonable.
